upup/pkg/fi/cloudup/awstasks/natgateway.go

```diff
@@ -209,19 +209,36 @@ func findNatGatewayFromRouteTable(cloud awsup.AWSCloud, routeTable *RouteTable)
 		}
 
 		if rt != nil {
+			clusterName, ok := routeTable.Tags[awsup.TagClusterName]
+			if !ok {
+				return nil, fmt.Errorf("Could not find '%s' tag from route table", awsup.TagClusterName)
+			}
 			var natGatewayIDs []*string
 			for _, route := range rt.Routes {
 				if route.NatGatewayId != nil {
 					natGatewayIDs = append(natGatewayIDs, route.NatGatewayId)
 				}
 			}
+			filteredNatGateways := []*ec2.NatGateway{}
+			for _, natGatewayID := range natGatewayIDs {
+				gw, err := findNatGatewayById(cloud, natGatewayID)
+				if err != nil {
+					return nil, err
+				}
```

Comment:
Because in the past there was no requirement for NatGW to be tagged in any way, one could use a shared VPC without any tags at all. Maybe this filtering logic should be used only when `len(natGatewayIDs) > 1`?